rust/crd/src/security.rs

```diff
@@ -0,0 +1,504 @@
+//! A helper module to process Apache Kafka security configuration
+//!
+//! This module merges the `tls` and `authentication` module and offers better accessibility
+//! and helper functions
+//!
+//! This is required due to overlaps between TLS encryption and e.g. mTLS authentication or Kerberos
+
+use crate::{
+    authentication, authentication::ResolvedAuthenticationClasses, listener, tls, KafkaCluster,
+    SERVER_PROPERTIES_FILE, STACKABLE_TMP_DIR,
+};
+
+use crate::listener::KafkaListenerConfig;
+use snafu::{ResultExt, Snafu};
+use stackable_operator::{
+    builder::{ContainerBuilder, PodBuilder, SecretOperatorVolumeSourceBuilder, VolumeBuilder},
+    client::Client,
+    commons::authentication::{AuthenticationClass, AuthenticationClassProvider},
+    k8s_openapi::api::core::v1::Volume,
+};
+use std::collections::BTreeMap;
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("failed to process authentication class"))]
+    InvalidAuthenticationClassConfiguration { source: authentication::Error },
+}
+
+/// Helper struct combining TLS settings for server and internal with the resolved AuthenticationClasses
+pub struct KafkaSecurity {
```

Comment:
Should we include opa in here as well?

Reply:
This is more of a grouper struct for TLS (e.g. on off depending on TLS / Auth settings) as well as which ports to use etc. 
OPA does not really fit here imho.

Reply:
Fine for me. Could we maybe rename this struct to e.g. `KafkaTlsSettings` or `ResolvedKafkaTlsSettings`?

Reply:
done